frontend/lib/firebase/firebase_authenticator.dart

```diff
@@ -1,15 +1,39 @@
 import 'package:firebase_auth/firebase_auth.dart';
 import 'package:google_sign_in/google_sign_in.dart';
-
-import '../domain/result.dart';
+import 'package:frontend/domain/result.dart';
 
 class FirebaseAuthenticator {
   final FirebaseAuth _auth = FirebaseAuth.instance;
   final GoogleSignIn _googleSignIn = GoogleSignIn();
 
   Future<Result<UserCredential>> signUp(String email, String password) async {
     try {
-      UserCredential userCredential = await _auth.createUserWithEmailAndPassword(email: email, password: password);
+      UserCredential userCredential = await _auth
+          .createUserWithEmailAndPassword(email: email, password: password);
+      return Result.success(userCredential);
+    } on FirebaseAuthException catch (e) {
+      return Result.failure(
+          ResultError(message: "Code:${e.code} Message:${e.message}"));
+    } catch (e) {
+      return Result.failure(ResultError(message: e.toString()));
+    }
+  }
+  
+  Future<Result<UserCredential>> signUpWithGoogle() async {
+    try {
+      final GoogleSignInAccount? googleUser = await _googleSignIn.signIn();
+      if (googleUser == null) {
+        return Result.failure(ResultError(message: "Google sign-in was cancelled by user"));
+      }
+
+      final GoogleSignInAuthentication googleAuth = await googleUser.authentication;
+
+      final credential = GoogleAuthProvider.credential(
+        accessToken: googleAuth.accessToken,
+        idToken: googleAuth.idToken,
+      );
+
+      UserCredential userCredential = await _auth.signInWithCredential(credential);
```

Comment:
:question: Is this the same implementation we have in `signInWithGoogle` method? Why we can't use previous implementation? We could change this method name to `authenticateWithGoogle` it it handles both sing in and sign up

Reply:
Yes, I changed this in #23 - we will use signInWithGoogle in both cases